nes-network/network/src/sender/control.rs

```diff
@@ -0,0 +1,453 @@
+            NetworkingConnectionControlMessage::RegisterChannel(channel, response) => {
+                let (sender, queue) = async_channel::bounded(100);
+                let channel_cancellation = CancellationToken::new();
+
+                tokio::spawn(
+                    attempt_channel_registration(
+                        this_connection.clone(),
+                        channel,
+                        channel_cancellation.clone(),
+                        queue,
+                        channel_registration_request_handler.clone(),
+                        controller.clone(),
+                        communication.clone(),
+                    )
+                        .in_current_span(),
+                );
+
+                active_channel.add_token(channel_cancellation);
+                let _ = response.send(sender);
+            }
+            NetworkingConnectionControlMessage::RetryChannel(
+                channel,
+                channel_cancellation,
+                queue,
+            ) => {
+                tokio::spawn(
+                    attempt_channel_registration(
+                        this_connection.clone(),
+                        channel,
+                        channel_cancellation.clone(),
+                        queue,
+                        channel_registration_request_handler.clone(),
+                        controller.clone(),
+                        communication.clone(),
+                    )
+                        .in_current_span(),
+                );
+            }
+        };
+    }
+}
+fn create_connection_handler(
+    this_connection: ThisConnectionIdentifier,
+    target_connection: ConnectionIdentifier,
+    communication: impl Communication + 'static,
+) -> (ScopedTask<()>, NetworkingConnectionController) {
+    let (tx, rx) = async_channel::bounded::<NetworkingConnectionControlMessage>(1024);
+    let control = tx.clone();
+    let task = ScopedTask::new(tokio::spawn(
+        {
+            let target_connection = target_connection.clone();
+            async move {
+                info!(
+                    "Connection is terminated: {:?}",
+                    connection_handler(
+                        this_connection,
+                        target_connection,
+                        control,
+                        rx,
+                        communication
+                    )
+                    .await
+                );
+            }
+        }
+            .instrument(info_span!(
+                parent: Span::current(),
+            "connection",
+            other = %target_connection
+        )),
+    ));
+    (task, tx)
+}
+
+pub(super) async fn network_sender_dispatcher(
+    this_connection: ThisConnectionIdentifier,
+    control: NetworkingServiceControlListener,
+    communication: impl Communication + 'static,
+) -> Result<()> {
+    // All currently active connections. Dropping the hashmap will abort all active connections.
```

Comment:
And that happens when this future here resolves/is dropped, which happens when the controller queue was closed from the outside?